src/components/App.tsx

```diff
@@ -920,16 +920,20 @@ class App extends React.Component<AppProps, AppState> {
     if (prevProps.langCode !== this.props.langCode) {
       this.updateLanguage();
     }
-
-    if (prevProps.viewModeEnabled !== this.props.viewModeEnabled) {
+    if (
+      prevProps.viewModeEnabled !== this.props.viewModeEnabled ||
+      prevState.viewModeEnabled !== this.state.viewModeEnabled
```

Comment:
why do we need to set state again when the state mismatches? this may trigger unnecessary rerender

Reply:
I wanted to merge those two branches of yours into one to guard against logic divergence — and I forgot to remove the 2nd branch afterwards.

Reply:
Nvm, I did remove the 2nd branch. Now I just mistakenly removed the zenMode branch :). Reverting.

Reply:
As for the unnecessary rerenders: there shouldn't be any on init. On later updates there may be but it shouldn't matter as it won't be on the hot path.

Reply:
I think we should not merge the two branches as it creates confusion as why set state when already set and if it happens such it will lead to rerender (looking at the code) so would be better to keep it separate for better readability IMO

Reply:
Ok. So what I did is to split the branches, but put everything relating to app state (apart from the `viewModeEnabled` prop itself) in the 2nd branch. This way there's no duplicate code that we need to keep in sync, and it's also less imperative.